File: 백준/Silver/2018. 수들의 합 5/수들의 합 5.cc
#include <iostream>
using namespace std;

int main(void)
{
	ios::sync_with_stdio(false);
	cin.tie(NULL);
	cin.tie(NULL);

	int start_idx = 1;
	int end_idx = 1;
	int sum = 1;
	int count = 1;
	int N;
	cin >> N;

	while (end_idx != N)
	{
		if (sum == N)
		{
			count++;
			end_idx++;
			sum += end_idx;
		}

		else if (sum < N)
		{
			end_idx++;
			sum += end_idx;
		}

		else if (sum > N)
		{
			sum -= start_idx;
			start_idx++;
		}
	}

	cout << count << "\n";

	return 0;
}

File: 백준/Silver/2018. 수들의 합 5/수들의 합 5.py
N = int(input())
sum = 1
count = 1
start_idx = 1
end_idx = 1

while end_idx != N:
    # 정답인 경우
    if sum == N:
        # 정답 카운트
        count += 1

        # 연속된 자연수의 범위를 한 칸 확장시킴
        end_idx += 1

        # 확장한 자연수를 합계에 추가
        sum += end_idx

    # 연속된 자연수의 합이 입력된 수보다 클 경우
    elif sum > N:
        # 연속된 자연수의 시작지점 인덱스를 합에서 뺀 후
        sum -= start_idx

        # 연속된 자연수에서 왼쪽 값 삭제
        start_idx += 1

    # 연속된 자연수의 합이 입력된 수보다 작을 경우
    else:
        # 연속된 자연수의 범위를 한 칸 확장시킴
        end_idx += 1

        # 확장한 자연수를 합계에 추가
        sum += end_idx

print(count)